src/rc.js

```diff
@@ -10,6 +10,14 @@ import * as rcUtil from './util/rc.js';
 // Keys that will get resolved relative to the path of the rc file they belong to
 const PATH_KEYS = new Set(['yarn-path', 'cache-folder', 'global-folder', 'modules-folder', 'cwd']);
 
+// prepares environment variables for processing rc
+export function prepareEnv() {
+  // Allow ${HOME} in, e.g., .npmrc to work even on Windows.
+  if (rcUtil.home && !process.env.HOME) {
+    process.env.HOME = rcUtil.home;
```

Comment:
I don't like this. We shouldn't overwrite environment variables and, if we really must, we shouldn't do it in this file.